primitives/src/merkle_tree/internal.rs

```diff
@@ -137,6 +137,9 @@ where
     pub pos: I,
     /// Nodes of proof path, from root to leaf
     pub proof: MerklePath<E, I, T>,
+    /// Root
+    #[serde(with = "canonical")]
+    pub root: T,
```

Comment:
I feel like the root value shouldn't be included in the Merkle proof structure.
The usual scenario is, party A would compare its locally maintained root value against the root value computed from party B's provided Merkle proof. It's weird if they are bind together.

Reply:
Makes sense. So I guess instead we should modify `verify` to take a new `root: T` argument? I'll do that.

Reply:
Done in https://github.com/EspressoSystems/jellyfish/pull/274/commits/57b35014d08d069d9a08a8566b146d93eb9a637c, https://github.com/EspressoSystems/jellyfish/pull/274/commits/2fb45633c60f09d733f5884201352886787cc858